pystan/model.py

```diff
@@ -64,10 +66,17 @@ def _map_parallel(function, args, n_jobs):
             import multiprocessing.pool
         except ImportError:
             multiprocessing = None
-        if sys.platform.startswith("win") and PY2:
-            msg = "Multiprocessing is not supported on Windows with Python 2.X. Setting n_jobs=1"
+        if (platform.system() == 'Windows') and PY2:
+            msg = 'Multiprocessing is not supported on Windows with Python 2.X. Setting n_jobs=1'
             logger.warning(msg)
             n_jobs = 1
+
```

Comment:
If we are going to introduce the practice of configuring things via environment variables, we should do it in a different pull request.

Reply:
What should we change in stanfit to enable threading? @wds15?

Reply:
I would recommend you to always compile with STAN_THREADS being defined.

You can stick to configuring things with the environment variable, but there are better options.

The crux is a bit that the thread configuration should better be handled in the stan services,  but that's still some time until we get that, I am afraid.

The better approach from my view is to use so-called task arena's from the TBB. These can be configured to allow some maximal concurrency and you run things within them. That's not too hard to do and it makes the environment variables obsolete which is what I would prefer as an interface dev.

Reply:
`STAN_THREADS` is called implemented as always on.

How would we automate task arena for a user? Do we need to start that first and then close it? Is it running as compiled C++ --> do we need to wrap it in python?

I would not want to add tbb-python as a requirement for pystan.

Reply:
I agree in that this should be part of the stan services... but that will likely take a moment.

So yes, you would have to instantiate a C++ object and run things with it... it's simple:

https://software.intel.com/en-us/node/506359

this is what should be done in the stan services layer somewhere, but we don't have it yet.

Reply:
@wds15 and @ahartikainen , how about waiting until this is all appropriately added to the stan services layer. We don't need to introduce bleeding edge stuff into pystan now that there are alternatives like cmdstanpy.

It would be a big accomplishment simply to get pystan updated to the current stan release. We don't need to do everything in this pull request.

@wds15 , is there an issue where we can track progress towards a more user-friendly way of dealing with the num_threads setting?

Reply:
That is a good idea.

Reply:
Sure. No there is no issue for that with the Stan repo. It will be a big effort to pull that off as it will likely impact a lot of the api. The issue is that knowledge on threads is with another set of people than that with people having a deep overview of the stan services...but that’s a different story.

Reply:
Having an issue to point to would be useful.

Can we hope that a sensible default might be chosen for people who don't have num threads set? That would be helpful in the short run. And I imagine it would be easier to implement.

Reply:
In the Short run we have Stan num Threads.

Reply:
I have now removed `n_threads` so user can still manually define `NUM_STAN_THREADS`.

`-DSTAN_THREADS` is still always used.